cvat/apps/consensus/models.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (C) 2024 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations  # this allows forward references
+
+from enum import Enum
+from typing import Any, Sequence
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.forms.models import model_to_dict
+
+from cvat.apps.engine.models import Job, ShapeType, Task, User
+
+
+class ConsensusConflictType(str, Enum):
+    NoMatchingItemError = "no_matching_item"
+    FailedAttrVotingError = "failed_attribute_voting"
+    NoMatchingAnnError = "no_matching_annotation"
+    AnnotationsTooCloseError = "annotation_too_close"
+    WrongGroupError = "wrong_group"
+    FailedLabelVotingError = "failed_label_voting"
+
+    def __str__(self) -> str:
+        return self.value
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+
+class AnnotationType(str, Enum):
+    TAG = "tag"
+    SHAPE = "shape"
+    TRACK = "track"
+
+    def __str__(self) -> str:
+        return self.value
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+
+class ConsensusReportTarget(str, Enum):
+    JOB = "job"
+    TASK = "task"
+
+    def __str__(self) -> str:
+        return self.value
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+
+class ConsensusSettings(models.Model):
+    task = models.ForeignKey(
+        Task,
+        on_delete=models.CASCADE,
+        related_name="consensus_settings",
+        null=True,
+        blank=True,
+    )
+    agreement_score_threshold = models.FloatField(default=0)
+    quorum = models.IntegerField(default=-1)
+    iou_threshold = models.FloatField(default=0.5)
+    sigma = models.FloatField(default=0.1)
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+        if self.quorum == -1:
+            self.quorum = self.task.consensus_jobs_per_regular_job // 2
```

Comment:
I think it's better to do either in serializer or in the using code. Otherwise, this field access can invoke extra db requests or this init call can be skipped if the object is created by other means.